src/main/scala/com/ing/wbaa/airlock/proxy/provider/atlas/LineageHelpers.scala

```diff
@@ -0,0 +1,129 @@
+      Seq(Classification(classification)))))
+
+  private def bucketEntities(userSTS: String, bucket: String, classification: String) =
+    Entities(Seq(Bucket(
+      "Bucket",
+      userSTS,
+      BucketAttributes(bucket, bucket, bucket),
+      Seq(Classification(classification)))))
+
+  private def fileEntities(
+      bucketGuid: String,
+      bucketObject: String,
+      userSTS: String,
+      contentType: ContentType,
+      classification: String) = {
+    Entities(Seq(IngestedFile(
+      "DataFile",
+      userSTS,
+      FileAttributes(bucketObject, bucketObject, bucketObject, contentType.mediaType.value,
+        guidRef(bucketGuid, "Bucket"),
+        Seq(Classification(classification))))))
+  }
+
+  private def processEntities(
+      serverGuid: String,
+      bucketGuid: String,
+      fileGuid: String,
+      userSTS: String,
+      method: String,
+      inputs: List[guidRef],
+      outputs: List[guidRef],
+      clientType: String,
+      timestamp: Long) = {
+    Entities(Seq(Ingestion(
+      "airlock_client",
+      userSTS,
+      IngestionAttributes(s"${clientType}_$timestamp", s"${clientType}_$timestamp", method, userSTS,
+        guidRef(serverGuid, "Server"),
+        inputs,
+        outputs))))
+  }
+
+  private def extractClient(userAgent: String): String =
+    """(\S+)/\S+""".r
+      .findFirstMatchIn(userAgent)
+      .map(_ group 1).getOrElse("generic")
+
+  private def extractHeaderOption(httpRequest: HttpRequest, header: String): Option[String] =
+    if (httpRequest.getHeader(header).isPresent)
+      Some(httpRequest.getHeader(header).get().value())
+    else
+      None
+
+  def getLineageHeaders(httpRequest: HttpRequest): LineageHeaders = {
+    val path = httpRequest.uri.path.toString().split("/")
+
+    LineageHeaders(
+      extractHeaderOption(httpRequest, "Remote-Address"),
+      path.take(path.length - 1).mkString("/"),
+      bucketObject = path.lastOption.getOrElse("emptyObject"),
+      httpRequest.method,
+      httpRequest.entity.contentType,
+      extractClient(extractHeaderOption(httpRequest, "User-Agent").getOrElse("generic")),
+      httpRequest.uri.rawQueryString,
+      extractHeaderOption(httpRequest, "x-amz-copy-source")
+    )
+  }
+
+  // file entity delete
+  // for now it is just deleting file entity and no related objects like eg. aws_cli_script, which uploaded or downloaded
+  // file to bucket. Once we delete aws_cli_script object we will lose track of whats has been deleted
+  // We need to come up with process of tracking file delete
+  def deleteEntities(typeName: String, entityName: String): Future[LineageGuidResponse] =
+    for {
+      entityGuidResponse <- getEntityGUID(typeName, entityName)
+      _ <- deleteEntity(entityGuidResponse.entityGUID)
+
+    } yield entityGuidResponse
+
+  def postEnities(
+      userSTS: String,
+      host: String,
+      bucket: String,
+      bucketObject: String,
+      method: String,
```

Comment:
As in other file, I'd make this type `AccessType`
```suggestion
      method: AccessType,
```